homeassistant/components/freedompro/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Support for Freedompro sensor."""
+from hatasmota.const import LIGHT_LUX, PERCENTAGE, TEMP_CELSIUS
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+)
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro sensor."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "temperatureSensor"
+        or device["type"] == "humiditySensor"
+        or device["type"] == "lightSensor"
+    )
+
+
+class Device(CoordinatorEntity, SensorEntity):
+    """Representation of an Freedompro sensor."""
+
+    def __init__(self, hass, api_key, device, coordinator):
+        """Initialize the Freedompro sensor."""
+        super().__init__(coordinator)
+        self._hass = hass
+        self._session = aiohttp_client.async_get_clientsession(self._hass)
+        self._api_key = api_key
+        self._attr_name = device["name"]
+        self._attr_unique_id = device["uid"]
+        self._type = device["type"]
+        self._characteristics = device["characteristics"]
+        self._attr_device_info = {
+            "name": self._attr_name,
+            "identifiers": {
+                (DOMAIN, self._attr_unique_id),
+            },
+            "model": self._type,
+            "manufacturer": "Freedompro",
+        }
+        self._attr_device_class = (
+            DEVICE_CLASS_TEMPERATURE
+            if self._type == "temperatureSensor"
+            else DEVICE_CLASS_HUMIDITY
+            if self._type == "humiditySensor"
+            else DEVICE_CLASS_ILLUMINANCE
+        )
```

Comment:
Add some constants instead, it make the code more readable.

```suggestion
DEVICE_CLASS_MAP = {
    "temperatureSensor": DEVICE_CLASS_TEMPERATURE,
    "humiditySensor": DEVICE_CLASS_HUMIDITY,
    "lightSensor": DEVICE_CLASS_ILLUMINANCE,
}
...
        self._attr_device_class = DEVICE_CLASS_MAP[self._type]
```